src/org/javarosa/core/model/instance/ExternalDataInstance.java

```diff
@@ -36,26 +36,27 @@ public ExternalDataInstance(String path, String instanceId, TreeElement root) {
         this.root = root;
     }
 
-    public static ExternalDataInstance buildFromPath(String path, String instanceId)
-            throws IOException, UnfullfilledRequirementsException, XmlPullParserException, InvalidStructureException {
-        String absolutePath = getPathPrefix() + path;
-        KXmlParser xmlParser = ElementParser.instantiateParser(new FileInputStream(absolutePath));
+    /**
+     * Builds an ExternaldataInstance
+     *
+     * @param path       the path to the XML file
```

Comment:
Maybe add "absolute" to the wording to make it explicit. It also would explain why line 41 was removed.